mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -251,7 +274,49 @@ def read_metrics_data(
             ],
         ],
     ]:
-        raise NotImplementedError
+        if type == "metrics":
+            table = mm_schemas.TDEngineSuperTables.METRICS
+            name = mm_schemas.MetricData.METRIC_NAME
+            df_handler = self.df_to_metrics_values
+        elif type == "results":
+            table = mm_schemas.TDEngineSuperTables.APP_RESULTS
+            name = mm_schemas.ResultData.RESULT_NAME
+            df_handler = self.df_to_results_values
+        else:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Invalid type {type}, must be either 'metrics' or 'results'."
+            )
+
+        metrics_condition = " OR ".join(
+            [
+                f"({mm_schemas.WriterEvent.APPLICATION_NAME} = '{metric.app}' AND {name} = '{metric.name}')"
+                for metric in metrics
+            ]
+        )
+        filter_query = f"endpoint_id='{endpoint_id}' AND ({metrics_condition})"
+
+        df = self.get_records(
+            table=table,
+            start=start,
+            end=end,
+            filter_query=filter_query,
+            timestamp_column=mm_schemas.WriterEvent.END_INFER_TIME,
+        )
+
+        df[mm_schemas.WriterEvent.END_INFER_TIME] = pd.to_datetime(
+            df[mm_schemas.WriterEvent.END_INFER_TIME]
+        )
+        df.set_index(mm_schemas.WriterEvent.END_INFER_TIME, inplace=True)
+
+        logger.debug(
+            "Converting a data-frame to a list of metrics or results values",
```

Comment:
```suggestion
            "Converting a DataFrame to a list of metrics or results values",
```

Reply:
I see it's referred to as a data-frame in a bunch of places. As far as I know, that's not the correct nomenclature.